lib/tasks/browserstack.rake

```diff
@@ -41,11 +41,11 @@ namespace :browserstack do
         # We're in a subprocess here - set the environment variable BS_BROWSER to the desired browser configuration.
         ENV['BS_BROWSER'] = browser.to_json
 
-        test_label = ['os', 'os_version', 'browser', 'browser_version'].map { |k| browser[k] }.join('_')
+        test_label = ['os', 'os_version', 'browser', 'browser_version'].map(&:browser).join('_')
 
         system("rspec spec/features --format RspecJunitFormatter --out '#{test_label}.xml'")
       end
-      exitstatus = results.count { |e| !e }
+      exitstatus = results.count(&:'!e')
```

Comment:
Surely this doesn't work - you'll just end up with a `NoMethodError`.

Anyways, the intention of this line seems to be to count the failures & if any exist cause the program to exit with failure. If we assume the 'bad' results are falsey values couldn't we just do:

```ruby
results.each { |result| abort unless result }
```
& do away with the ensure as well 😕